src/ripple/app/tx/impl/SetAccount.cpp

```diff
@@ -226,7 +241,7 @@ SetAccount::doApply ()
     //
     if ((uSetFlag == asfDisableMaster) && !(uFlagsIn & lsfDisableMaster))
     {
-        if (!mSigMaster)
+        if (!sigWithMaster)
```

Comment:
Let me start out by saying that the change you made here is great.  A piece of validation that was being done for every transaction has been localized to the only transaction type that needed it.  Very nice.

I have a hair-brained idea for a possible minor optimization.  Now that the check is localized to SetAccount, we might be able to localize it further to only those two sub-functions of SetAccount that actually need it: DisableMaster and SetNoFreeze.  Suppose you turned the code that computes `sigWithMaster` into a lambda.  Then you could call that lambda here and in the SetNoFreeze case, rather than pre-computing the bool and checking the bool here.

I'm not completely confident this is a good idea.  If there's an AccountSet call that does both DisableMaster and SetNoFreeze, then the computation is done twice, which would be a pessimization.  And optimizing this particular corner of SetAccount is probably unimportant in the larger scheme of things.  I just thought I'd mention the idea.